### PR TITLE
Fix #3781: Glitchy map causing crash

### LINKED ISSUE #3781: Glitchy map causing crash
**OS:** Windows
**Commit:** fbcc68d

Taken from https://openrct2.org/forums/topic/1332-cant-load-server/

> The game either crashes or says "Unable to load file: _random characters_". 

The map looks glitchy, there's lots of

```
ERROR[/home/janisozaur/workspace/OpenRCT2/src/game.c:766 (game_fix_save_vars)]: Null map element at x = 134 and y = 255. Fixing...
ERROR[/home/janisozaur/workspace/OpenRCT2/src/game.c:766 (game_fix_save_vars)]: Null map element at x = 136 and y = 255. Fixing...
ERROR[/home/janisozaur/workspace/OpenRCT2/src/game.c:766 (game_fix_save_vars)]: Null map element at x = 141 and y = 255. Fixing...
ERROR[/home/janisozaur/workspace/OpenRCT2/src/game.c:766 (game_fix_save_vars)]: Null map element at x = 145 and y = 255. Fixing...
ERROR[/home/janisozaur/workspace/OpenRCT2/src/game.c:766 (game_fix_save_vars)]: Null map element at x = 152 and y = 255. Fixing...
ERROR[/home/janisozaur/workspace/OpenRCT2/src/game.c:766 (game_fix_save_vars)]: Null map element at x = 164 and y = 255. Fixing...
```

**Save game:**
[a764ce64-072a-48a8-9969-e8c9cc2f38c6.sv6.txt](https://github.com/OpenRCT2/OpenRCT2/files/290121/a764ce64-072a-48a8-9969-e8c9cc2f38c6.sv6.txt)

**Backtrace:**

```
>>> p entry
$1 = (rct_scenery_entry *) 0xffffffff
>>> bt
#0  0x015568ea in scenery_paint (direction=1 '\001', height=160, mapElement=0x102e5e0) at /home/janisozaur/workspace/OpenRCT2/src/paint/map_element/scenery.c:67
#1  0x015543e7 in sub_68B3FB (x=5792, y=3872) at /home/janisozaur/workspace/OpenRCT2/src/paint/map_element/map_element.c:240
#2  0x01553ec2 in map_element_paint_setup (x=5792, y=3840) at /home/janisozaur/workspace/OpenRCT2/src/paint/map_element/map_element.c:58
#3  0x0155de68 in viewport_paint_setup () at /home/janisozaur/workspace/OpenRCT2/src/paint/paint.c:707
#4  0x0149e29a in viewport_paint (viewport=0x179dec0 <g_viewport_list>, dpi=0xffffcdb4, left=9672, top=988, right=11000, bottom=1016) at /home/janisozaur/workspace/OpenRCT2/src/interface/viewport.c:789
#5  0x0149df2f in viewport_render (dpi=0xffffcdb4, viewport=0x179dec0 <g_viewport_list>, left=9672, top=988, right=11000, bottom=1016) at /home/janisozaur/workspace/OpenRCT2/src/interface/viewport.c:691
#6  0x0142abac in window_main_paint (w=0x178aea0 <g_window_list>, dpi=0xffffcdb4) at /home/janisozaur/workspace/OpenRCT2/src/windows/main.c:98
#7  0x0149b399 in window_event_paint_call (w=0x178aea0 <g_window_list>, dpi=0xffffcdb4) at /home/janisozaur/workspace/OpenRCT2/src/interface/window.c:1974
#8  0x0149a5b2 in window_draw (w=0x178aea0 <g_window_list>, left=0, top=0, right=1328, bottom=28) at /home/janisozaur/workspace/OpenRCT2/src/interface/window.c:1543
#9  0x0150a1b4 in gfx_redraw_screen_rect (left=0, top=0, right=1328, bottom=28) at /home/janisozaur/workspace/OpenRCT2/src/drawing/drawing.c:355
#10 0x0149d457 in sub_6E7F34 (w=0x178b360 <g_window_list+1216>, viewport=0x179dec0 <g_viewport_list>, x_diff=-18526, y_diff=-1456) at /home/janisozaur/workspace/OpenRCT2/src/interface/viewport.c:426
#11 0x0149d873 in sub_6E7DE1 (x=9672, y=988, w=0x178aea0 <g_window_list>, viewport=0x179dec0 <g_viewport_list>) at /home/janisozaur/workspace/OpenRCT2/src/interface/viewport.c:500
#12 0x0149dc98 in viewport_update_position (window=0x178aea0 <g_window_list>) at /home/janisozaur/workspace/OpenRCT2/src/interface/viewport.c:628
#13 0x01497979 in window_update_all_viewports () at /home/janisozaur/workspace/OpenRCT2/src/interface/window.c:149
#14 0x014979ac in window_update_all () at /home/janisozaur/workspace/OpenRCT2/src/interface/window.c:161
#15 0x01569ebf in game_load_init () at /home/janisozaur/workspace/OpenRCT2/src/game.c:866
#16 0x01569d25 in game_load_save (path=0x17672e0 <gOpenRCT2StartupActionPath> "/home/janisozaur/Downloads/a764ce64-072a-48a8-9969-e8c9cc2f38c6.sv6") at /home/janisozaur/workspace/OpenRCT2/src/game.c:812
#17 0x015610ab in rct2_open_file (path=0x17672e0 <gOpenRCT2StartupActionPath> "/home/janisozaur/Downloads/a764ce64-072a-48a8-9969-e8c9cc2f38c6.sv6") at /home/janisozaur/workspace/OpenRCT2/src/rct2.c:358
#18 0x01552bfb in openrct2_launch () at /home/janisozaur/workspace/OpenRCT2/src/openrct2.c:281
#19 0x0150edf2 in main (argc=2, argv=0xffffd374) at /home/janisozaur/workspace/OpenRCT2/src/platform/posix.c:60
```

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/openrct2/world/map.c [hunk 1]
```diff
@@ -3545,6 +3545,8 @@ sint32 map_element_get_banner_index(rct_map_element *mapElement)
             ((mapElement->properties.scenerymultiple.colour[1] & 0xE0) >> 5);
     case MAP_ELEMENT_TYPE_WALL:
         sceneryEntry = get_wall_entry(mapElement->properties.wall.type);
+        if (sceneryEntry == NULL)
```
Comment (rwjuk) [inline src/openrct2/world/map.c]: Why not combine the conditions into `if (sceneryEntry == NULL || sceneryEntry->wall.scrolling_mode == 0xFF)`?

Comment (janisozaur) [reply src/openrct2/world/map.c]: I did that in other places, but missed that one

Comment (janisozaur) [reply src/openrct2/world/map.c]: done

**Comment by Gymnasiast** [comment]
That map is so enormously damaged, I'm surprised those null guards are enough.

**Pull request merged**